src/translate/pddl_parser/parsing_functions.py

```diff
@@ -313,186 +547,168 @@ def parse_task(domain_pddl, task_pddl):
         predicates, functions, init, goal, actions, axioms, use_metric)
 
 
-def parse_domain_pddl(domain_pddl):
+def parse_domain_pddl(context, domain_pddl):
     iterator = iter(domain_pddl)
+    with context.layer("Parsing domain"):
+        define_tag = next(iterator)
+        if define_tag != "define":
+            context.error("Domain definition expected to start with '(define ")
```

Comment:
Alternative suggestion to be closer to victor's suggested message in issue1029:
f"Domain definition expected to start with '(define', but got ({define_tag}"

It's a bit hacky with the parenthesis before {define_tag}, but if the parenthesis were missing we would catch it at an earlier place.